services/worker/src/worker/job_runners/dataset/compatible_libraries.py

```diff
@@ -607,20 +619,22 @@ def compute_compatible_libraries_response(
                 "Previous step 'dataset-info' did not return the expected content.", e
             ) from e
     if infos:
-        # mlcroissant library
-        libraries.append(get_mlcroissant_compatible_library(dataset, infos, partial=partial))
-        tags.append("croissant")
         # datasets library
         libraries.append(get_hf_datasets_compatible_library(dataset, infos))
         # pandas or dask or webdataset library
         builder_name = infos[0]["builder_name"]
+        if builder_name in get_format_for_builder:
+            formats.append(get_format_for_builder[builder_name])
         if builder_name in get_compatible_library_for_builder:
             try:
                 compatible_library = get_compatible_library_for_builder[builder_name](dataset, hf_token)
                 libraries.append(compatible_library)
             except NotImplementedError:
                 pass
-    return DatasetCompatibleLibrariesResponse(tags=tags, libraries=libraries)
+        # mlcroissant library
+        libraries.append(get_mlcroissant_compatible_library(dataset, infos, partial=partial))
+        tags.append("croissant")
```

Comment:
good idea to move them to the end of the libraries list (as we will respect this order on the Hub)